packages/admin-ui/src/components/Filter/components/FilterCheckbox/useFilterCheckbox.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useCallback, useMemo, useRef, useEffect } from 'react'
+import type { UseFilterStateReturn, FilterItem } from '../Filter'
+import { usePopoverState } from '../Popover'
+import { Item } from '@react-stately/collections'
+
+import { useListBox } from '@react-aria/listbox'
+import type { ListState } from '@react-stately/list'
+import { useListState } from '@react-stately/list'
+
+export function useFilterCheckbox<T extends FilterItem>(
+  props: UseFilterCheckboxStateProps<T>
+): UseFilterCheckboxReturn<T> {
+  const { onApply, items, label } = props
+  const [selectedKeys, setSelectedKeys] = useState<key[]>([])
+  const popover = usePopoverState({ gutter: 0, placement: 'bottom-start' })
+
+  const stateProps = {
+    items,
+    children: (item: FilterItem) => <Item>{item.label}</Item>,
+    selectionMode: 'multiple' as const,
+  }
+
+  const ref = useRef(null)
+
+  const listState = useListState<T>(stateProps)
+
+  const { listBoxProps, labelProps } = useListBox<FilterItem>(
+    { ...stateProps, 'aria-label': label },
+    listState,
+    ref
+  )
+
+  useEffect(() => {
+    if (!popover.visible) {
+      listState.selectionManager.setSelectedKeys(selectedKeys)
+    }
+  }, [popover.visible])
+
+  const selectedValues = useMemo(
+    () => selectedKeys.map((k) => listState.collection.getItem(k).value.label),
+    [selectedKeys, listState.collection]
+  )
+
+  const apply = useCallback(() => {
+    const nextSelected = [...listState.selectionManager.selectedKeys]
+
+    setSelectedKeys(nextSelected)
+    onApply?.({ selected: nextSelected })
```

Comment:
The onApply here is interesting because it runs in both apply and clear functions. Maybe rename it to `onChangeSelected`?